src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
It's not really "different keymap per window", but rather "the compositor advertizes the current keymap to the clients, but the clients are free to use their own". The `wl_keyboard::keymap` event is only informative. Client are free to use their own keymap config.